src/plugins/intel_gpu/src/graph/impls/ocl/convolution.cpp

```diff
@@ -278,7 +278,28 @@ struct convolution_impl : typed_primitive_impl_ocl<convolution> {
 
 std::unique_ptr<primitive_impl> ConvolutionImplementationManager::create_impl(const program_node& node, const kernel_impl_params& params) const {
     OPENVINO_ASSERT(node.is_type<convolution>());
-    return typed_primitive_impl_ocl<convolution>::create<convolution_impl>(static_cast<const convolution_node&>(node), params);
+    auto impl = typed_primitive_impl_ocl<convolution>::create<convolution_impl>(static_cast<const convolution_node&>(node), params);
+
+    if (auto weights_reorder_params = impl->get_weights_reorder_params()) {
+        auto input_layout = params.get_input_layout(0);
+        if (input_layout.get_partial_shape().size() == 3) {
+            auto weights_input_layout = weights_reorder_params->get_input_layout();
+            if (weights_input_layout.group() == 1) {
+                weights_input_layout.set_partial_shape(ov::PartialShape{weights_input_layout.batch(),
+                                                                        weights_input_layout.feature(),
+                                                                        weights_input_layout.spatial(1) * weights_input_layout.spatial(0)});
+                                                                        weights_reorder_params->set_input_layout(weights_input_layout);
+
+                auto weights_output_layout = weights_reorder_params->get_output_layout();
+                weights_output_layout.set_partial_shape(ov::PartialShape{weights_output_layout.batch(),
+                                                                         weights_output_layout.feature(),
+                                                                         weights_output_layout.spatial(1) * weights_output_layout.spatial(0)});
+                                                                         weights_reorder_params->set_output_layout(weights_output_layout);
+            }
```

Comment:
And only grouped format case would be fixed with this change?

Reply:
Also can't we just use the original Partial shape? (kernel_impl_params' shape)
I think weight reorder is enough to change the format, not the shape here.

Reply:
> And only grouped format case would be fixed with this change?

No, the change is used only for non group case, as grouped case, some tests failed.

> Also can't we just use the original Partial shape? (kernel_impl_params' shape) I think weight reorder is enough to change the format, not the shape here.

I tired but it seemed to work only for output layout. And as I replied to Sergey's comment above, I tried to use pop_back() to reduce 1d in shape and updated PR.
If I did anything wrong, let me know.

Reply:
It is updated to use original partial shape as you commented.
Thanks @yeonbok and @sshlyapn.